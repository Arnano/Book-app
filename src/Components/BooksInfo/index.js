import React from 'react';
import PropTypes from 'prop-types';
import StyledBooksInfo from './style';

import Section from '../Section';
import Text from '../Text';
import Heading from '../Heading';

const BooksInfo = ({ author, category, classes, description, rating, title }) => 
  <StyledBooksInfo className={classes} >
    <Section>
      <Heading level={2} size='h2'> {title} </Heading>
      <Heading level={3} size='h3'> {author} </Heading>
      { rating && <Text size='fsMedium'> rating: {rating} </Text> }
    </Section>
    <Section alignment='center' >
      { description && <p> <Text size='fsSmall'> {description} </Text> </p> }
      { category && <Text size='fsSmall'> Category: {category} </Text> }
    </Section>
  </StyledBooksInfo >

BooksInfo.propTypes = {
  author: PropTypes.string.isRequired,
  category: PropTypes.string,
  classes: PropTypes.string,
  description: PropTypes.string.isRequired,
  rating: PropTypes.number,
  title: PropTypes.string.isRequired
}

export default BooksInfo;
