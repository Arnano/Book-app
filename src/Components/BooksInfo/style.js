import Styled from 'styled-components';

const StyledBooksInfo = Styled.div `
`;

export default StyledBooksInfo;
