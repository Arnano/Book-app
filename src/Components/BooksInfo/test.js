import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import BooksInfo from './';

Enzyme.configure({
  adapter: new Adapter()
}); 

describe('Books info', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BooksInfo 
        author={'Arnaud'}
        category={'Fantasy'}
        description={'Book about fantasy'}
        rating={3}
        title={'Terrendiul'} /> , div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <BooksInfo 
      author={'Arnaud'}
      category={'Fantasy'}
      description={'Book about fantasy'}
      rating={3}
      title={'Terrendiul'} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

