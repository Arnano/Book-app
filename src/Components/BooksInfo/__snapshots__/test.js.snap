// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books info has a valid snapshot 1`] = `
<div
  className="sc-bdVaJa fePDys"
>
  <div>
    <h1
      className="sc-ifAKCX cHGRBc"
      color="darkGrey"
      id={undefined}
      size="h1"
    >
       
      Terrendiul
       
    </h1>
    <h2
      className="sc-EHOje boPmQa"
      color="darkGrey"
      id={undefined}
      size="h2"
    >
       
      Arnaud
       
    </h2>
    <span
      className="sc-bwzfXH iVZgoT"
      color="darkGrey"
      size="fsLarge"
    >
       rating: 
      3
       
    </span>
  </div>
  <p>
     
    <span
      className="sc-bwzfXH fkrTKN"
      color="darkGrey"
      size="fsMedium"
    >
       
      Book about fantasy
       
    </span>
     
  </p>
  <span
    className="sc-bwzfXH fkrTKN"
    color="darkGrey"
    size="fsMedium"
  >
     Category: 
    Fantasy
     
  </span>
</div>
`;
